### ISSUE TITLE
Running npm rebuild always reinstalls the binary

### ISSUE THREAD

**Issue by kudos** [Medium/phantomjs#255]
We use phantomjs as a dependency via karma, but we don't actually use it. Whenever we rebuild our node modules (like on travis), phantomjs goes and downloads the binary and installs it, even though the binary is already present through caching of the node_modules directory.

It seems that if you can't find a global binary, you just go ahead and do the install process again even though it already exists in the packages .bin directory.

**Comment by sergiopereiraTT** [comment]
So does it mean we are stuck without a fix or recommended work-around for this problem? I'm open to work-arounds since the comments linked above are discouraging enough to try and submit a patch without understanding all the use cases.

I'm running into the same issue in my Jenkins job. Every time the job runs it tries to make sure *karma-phantomjs-launcher* is installed and that ends up downloading and building phantomjs:

```
# running "npm install karma-phantomjs-launcher"

> phantomjs@1.9.16 install /path/to/job-workspace/node_modules/karma-phantomjs-launcher/node_modules/phantomjs
> node install.js

Looks like an `npm install -g`; unable to check for already installed version.
Downloading https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-1.9.8-linux-x86_64.tar.bz2
Saving to /path/to/job-workspace/node_modules/karma-phantomjs-launcher/node_modules/phantomjs/phantomjs/phantomjs-1.9.8-linux-x86_64.tar.bz2
Receiving...

Received 12854K total.
Extracting tar contents (via spawned process)
Removing /path/to/job-workspace/node_modules/karma-phantomjs-launcher/node_modules/phantomjs/lib/phantom
Copying extracted folder /path/to/job-workspace/node_modules/karma-phantomjs-launcher/node_modules/phantomjs/phantomjs/phantomjs-1.9.8-linux-x86_64.tar.bz2-extract-1427820425304/phantomjs-1.9.8-linux-x86_64 -> /path/to/job-workspace/node_modules/karma-phantomjs-launcher/node_modules/phantomjs/lib/phantom
Writing location.js file
Done. Phantomjs binary available at /path/to/job-workspace/node_modules/karma-phantomjs-launcher/node_modules/phantomjs/lib/phantom/bin/phantomjs
karma-phantomjs-launcher@0.1.4 node_modules/karma-phantomjs-launcher
└── phantomjs@1.9.16 (which@1.0.9, progress@1.1.8, kew@0.4.0, request-progress@0.3.1, adm-zip@0.4.4, npmconf@2.1.1, fs-extra@0.16.5, request@2.42.0)
```

**Comment by nicks** [comment]
if you control the machine, you can just install phantomjs on PATH and it will automatically use that.

**Comment by nicks** [comment]
This bug is particularly for the case where you're using `rebuild` and you want to be a little bit smarter. This is _not_ for the case where you're using `install` because there's already a solution for that.

see also: https://github.com/Medium/phantomjs#i-tried-everything-but-my-network-is-b0rked-what-do-i-do

**Comment by sergiopereiraTT** [comment]
I'm not sure "my network is b0rked" represents my problem but I appreciate
the suggestion to install in the PATH.

If memory doesn't fail me, I think doing that caused karma to fail unless I
also install it globaw, which I fear could interfere with other projects
that share the same build agent. I'll try again anyway.

--
Sergio Pereira, x6380, m: (224) 627-8352

**Comment by nicks** [comment]
if you are having a problem getting the PATH solution to work, please file
a separate bug, rather than adding noise to this bug.

**Comment by sergiopereiraTT** [comment]
@nicks sorry. That was not my intention.

**Comment by M0ns1gn0r** [comment]
@nicks 
>This is not for the case where you're using install because there's already a solution for that.

Could you please describe this solution?

**Comment by vipuldelwadia** [comment]
I too would like to know how to stop phantomjs from being installed each time you call `npm install`

**Comment by johanneslumpe** [comment]
@M0ns1gn0r @vipuldelwadia There is a link right above your comments: https://github.com/Medium/phantomjs#i-tried-everything-but-my-network-is-b0rked-what-do-i-do

TLDR: Install phantomjs locally and put it in your `PATH`

**Comment by vipuldelwadia** [comment]
@johanneslumpe That's not a great solution really, more of a workaround. Is there a plan for a "proper" fix?

**Comment by johanneslumpe** [comment]
@vipuldelwadia I do not know that - but a workaround is better than nothing, right?
